src/features/welcome/setting-welcome-page.test.tsx

```diff
@@ -0,0 +1,99 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+
+import type { RenderResult } from "@testing-library/react";
+import React from "react";
+import type { ApplicationBuilder } from "../../renderer/components/test-utils/get-application-builder";
+import { getApplicationBuilder } from "../../renderer/components/test-utils/get-application-builder";
+import type { FakeExtensionOptions } from "../../renderer/components/test-utils/get-extension-fake";
+import welcomeRouteConfigInjectable from "../../common/front-end-routing/routes/welcome/welcome-route-config.injectable";
+import welcomeRouteInjectable from "../../common/front-end-routing/routes/welcome/welcome-route.injectable";
+import type { Route } from "../../common/front-end-routing/front-end-route-injection-token";
+
+
+describe("setting-welcome-page", () => {
+  let applicationBuilder: ApplicationBuilder;
+  let rendered : RenderResult;
+  let welcomeRoute: Route;
+
+  beforeEach(() => {
+    applicationBuilder = getApplicationBuilder();
+  });
+
+  describe("given no extension sets the welcome page", () => {
```

Comment:
I would change this to read "given configuration of welcome page route is the default" because it is not necessarily an extension changing it and more the process of building.